lib/events.js

```diff
@@ -162,38 +323,62 @@ EventEmitter.prototype.emit = function emit(type) {
     return false;
   }
 
-  handler = events[type];
+  const handler = events[type];
 
-  if (!handler)
+  if (handler === undefined)
     return false;
 
-  if (domain && this !== process) {
+  if (domain != null && this !== process) {
```

Comment:
Looks like a possible behavior change. Possibly not documented, and not supported, but a change none the less. I'd like to see these broken out into a separate commit since they aren't actually related to the focal point of the changes going on here.